.github/workflows/pr.yml

```diff
@@ -16,16 +16,12 @@ jobs:
 
     runs-on: ${{ matrix.os }}
     steps:
-      - name: Setup Go
-        uses: actions/setup-go@v3
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
         with:
           go-version: 1.17
```

Comment:
1.17 is getting old, could we either bump this one or matrix (like we for os) it to test more modern go versions as well maybe?